libsplinter/src/biome/rest_api/actix/profile.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use crate::actix_web::HttpResponse;
+use crate::futures::IntoFuture;
+use crate::protocol;
+use crate::rest_api::{
+    ErrorResponse, HandlerFunction, Method, ProtocolVersionRangeGuard, Resource,
+};
+
+#[cfg(feature = "biome-profile")]
+use crate::biome::profile::store::{UserProfileStore, UserProfileStoreError};
+
+#[cfg(feature = "authorization")]
+use crate::biome::rest_api::BIOME_USER_READ_PERMISSION;
+
+/// Defines a REST endpoint to list profiles from the database
+pub fn make_profile_list_route(profile_store: Arc<dyn UserProfileStore>) -> Resource {
```

Comment:
We've recently started breaking up our REST API endpoints into different modules based on the endpoint, so these two `make_profile_*` methods should be in separate files (see the admin or registry REST APIs for reference).